drivers/net/phy/realtek.c

```diff
@@ -819,7 +951,36 @@ static int rtl8221b_vb_cg_match_phy_device(struct phy_device *phydev)
 		id |= val;
 	}
 
-	return (id == 0x001cc849);
+	if (id != RTL_8221B_VB_CG) return 0;
+
+	if (phydev->is_c45)
+		phydev->c45_ids.device_ids[1] = id;
+	else
+		phydev->phy_id = id;
+
+	return true;
+}
+
+/* Only use Clause 45 if available and Clause 22 is not.
+* This also includes RollBall protocol SFP modules.
+* Could possibly replace this function call with phydev->is_c45
+* to use the Clause 45 driver instance on all capable rtl822x.
+*/
+static bool rtl822x_use_c45(struct phy_device *phydev)
+{
+        return !phydev->mdio.bus->read && !!phydev->mdio.bus->read_c45;
```

Comment:
Let's just use `phydev->is_c45` here like you are suggesting in the comment

Reply:
Ok, then can remove the whole `static bool rtl822x_use_c45(struct phy_device *phydev) { }`

Reply:
in rtl8221b_vb_cg_match_phy_device()

You use if (phydev->mdio.bus->read_c45) to check if it can use phy_read_mmd() phy_write_mmd()

How about using the here too?